src/include/units/bits/basic_concepts.h

```diff
@@ -246,6 +246,32 @@ struct _point_kind_base;
 template<typename T>
 concept PointKind = kind_impl_<T, detail::_point_kind_base>;
 
+// PointOrigin
+namespace detail {
+
+template <typename>
+struct _origin_base;
+
+}  // namespace detail
+
+template<typename T, template<typename...> typename Base>
+concept origin_impl_ =
+is_derived_from_specialization_of<T, Base> &&
+requires(T* t) {
+  typename T::base_origin;
```

Comment:
Are there any constraints on the `base_origin` type itself (i.e. should it be a `Quantity`)?

Reply:
Actually, it should be a `PointOrigin` too. Actually, here, the convention is that the `base_origin` is an origin without any further `base_origin` set (which makes it some sort of `BasePointOrigin`). However, the traits machinery below would probably be simpler if the hypothetical `BasePointOrigin` were instead characterised by a recursive `base_origin` pointing to itself.

In general, this here is a somewhat experimental implementation to enable the implementation of related origins. In this design, an origin is simply a tag identifying the abstract concept of that origin. Then, we need a way to specify the "difference" between related origins, only where it is known and fixed. In order to avoid O(n^2) implementation complexity, currently all related origins need to agree on a reference/base origin.

Reply:
Yes. The additional checks are done below, in the non-exposition only concept.